controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -706,38 +743,51 @@ func (r *StorageClassClaimReconciler) reconcileCephBlockPool() error {
 
 	capacity := r.storageConsumer.Spec.Capacity.String()
 
+	cephClients := map[string]string{
+		"provisioner": r.cephClientProvisioner.Name,
+		"node":        r.cephClientNode.Name,
+	}
+
 	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.cephBlockPool, func() error {
 		if err := r.own(r.cephBlockPool); err != nil {
 			return err
 		}
+		deviceClass := r.deviceClass
+		parameters := r.blockPoolConfiguration.Parameters
 		deviceSetList := r.storageCluster.Spec.StorageDeviceSets
 		var deviceSet *v1.StorageDeviceSet
 		for i := range deviceSetList {
 			ds := &deviceSetList[i]
-			if ds.Name == "default" {
+			// To support blockpool created in 4.11
+			if ds.Name == "default" && deviceClass == "" {
+				deviceSet = ds
+				break
+			}
+			// get the required deviceSetName of the profile
+			if deviceClass == ds.DeviceClass {
 				deviceSet = ds
 				break
 			}
 		}
+		if deviceSet == nil && deviceClass == "" {
+			return fmt.Errorf("could not find device set named default in Storage cluster")
+		}
```

Comment:
Why do we care about the name ("default")?
This code should also apply for any new block pool created without a device set, as this is an option already available on the storage cluster.

A better way to think about this is to start with the profile itself. As we already established a profile will always have a device class field set on it. This means that r.deviceClass cannot be empty which means we should not concern ourselves with the "default" deviceClass. 

When talking about an "upgrade" scenario, we need to take into account that the default profile might want to refer to the existing block pool. In that case we have two options:!
1. add a deviceClass to the existing device set (if possible)
2. have the default profile refer to a deviceClass names "" (empty string)

Reply:
Yes i was thinking about the upgrades, I will remove this check for now